docs/build/frames.md

```diff
@@ -0,0 +1,345 @@
+    }
+  } else if (key.startsWith(IMAGE_PREFIX)) {
+    image = extractedTags[key];
+  } else if (key.startsWith(POST_URL_PREFIX)) {
+    postUrl = extractedTags[key];
+  } else if (key.startsWith(TITLE_PREFIX)) {
+    title = extractedTags[key];
+  }
+
+  // Separately handle action tags to fill the actions object and directly assign to buttons
+  if (key.includes(":action")) {
+    const actionIndex = parseInt(key.split(":")[3], 10) - 1; // Adjusted to match buttonIndex calculation
+    // Initialize the button object if it doesn't exist
+    buttons[actionIndex] = buttons[actionIndex] || {};
+    // Set the action for the button
+    buttons[actionIndex].action = extractedTags[key];
+  }
+}
+```
+
+### Rendering Frames
+
+Frames are rendered by dynamically creating a `Frame` component based on the extracted frame information. This component is responsible for displaying the frame's content, including images, titles, and interactive buttons.
+
+```jsx
+//word is the extracted frame url
+const framesClient = new FramesClient(client);
+const metadata = await framesClient.proxy.readMetadata(word);
+if (metadata) {
+  const info = getFrameInfo(metadata.extractedTags);
+  console.log("Frame info: ", info);
+  info.url = metadata.url;
+  setFrameInfo(info);
+}
+```
+
+Once we have the frame metadata we can render them visually
+
+```jsx
+{
+  !isLoading && frameInfo && (
+    <Frame
+      info={frameInfo}
+      handleClick={handleFrameButtonClick}
+      frameButtonUpdating={frameButtonUpdating}
+      frameUrl={frameInfo.url} // Passing the new prop
+    />
+  );
+}
+```
+
+Finally, we render the buttons dynaically
+
+```jsx
+{buttons?.map((button, index) => {
+if (!button) {
+    return null;
+    }
+    const handlePress = () => handleClick(index + 1);
+        const buttonStyle = {
+        ...styles.button,
+        ...(index === 0 ? styles.firstButton : {}),
+        ...(index === buttons.length - 1 ? styles.lastButton : {}),
+        };
+        return (
+        <button
+            key={`${button}-${index}`}
+            onClick={handlePress}
+            disabled={frameButtonUpdating === index + 1}
+            style={buttonStyle}>
+            {button.label}{" "}
+            {button.action === "post_redirect" && (
+            <span style={styles.redirectIcon}>↪</span>
+            )}
+        </button>
+);
+```
+
+### Handling Clicks in the Frames
+
+Clicks within the frame are handled by the `handleFrameButtonClick` function. This function determines the action to be taken based on the button clicked within the frame.
+
+```jsx
+  const handleFrameButtonClick = async (buttonNumber) => {
+    if (!frameInfo) {
+      return;
+    }
+    const actionType = frameInfo.buttons[buttonNumber - 1].action;
+    console.log("Action type: ", actionType);
+    console.log(frameInfo);
+
+    const conversationTopic = message.conversationTopic;
+
+    //Continue with Frames logic
+    ...
+```
+
+### Sign XMTP Payload
```

Comment:
```suggestion
### Signing the XMTP Payload
```

To conform with Validating/Getting/Rendering/Handling verbs from previous section titles.